google/cloud/pubsub_v1/publisher/client.py

```diff
@@ -399,8 +399,10 @@ def on_publish_done(future):
                     transport = self._transport
                     base_retry = transport._wrapped_methods[transport.publish]._retry
                     retry = base_retry.with_deadline(2.0**32)
+                    timeout = 2.0**32
```

Comment:
Add comment to explain what parameter this is. Add comment to explain why we set this to infinite (not obvious).